src/jshint.js

```diff
@@ -863,7 +863,7 @@ var JSHINT = (function() {
       funct["(blockscope)"].stack();
       advance("let");
       advance("(");
-      state.syntax["let"].fud.call(state.syntax["let"].fud, false);
+      state.syntax["let"].fud.call(state.syntax["let"].fud);
```

Comment:
The problem is if any of these constructs ever mutate their `this` binding, then you end up with problems if calling them that way, so I would prefer we didn't do that